### PR TITLE
Phases

### PR DESCRIPTION
**Pull request by mchalapuk**
This PR adds `markdox.parse`, `markdox.format`, and `markdox.render` functions to module's API.

* Whole code is rewritten, added some tests.
* I've also made some changes to README.md and .editorconfig.

If you decide to merge this, I will deprecate gulp-markdox2 on npm with a redirect to gulp-markdox.

Cheers!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,29 +7,32 @@
...
+    "ejs",
+    "dox"
   ],
   "author": {
     "name": "Guilherme Berger",
     "email": "guilherme.berger@gmail.com",
     "url": "https://github.com/gberger"
   },
-  "repository": "gberger/gulp-markdox",
+  "repository": "gberger/gulp-markdox2",
```
Comment (gberger) [inline package.json]: please change back to gulp-markdox

File: package.json [hunk 2]
```diff
@@ -7,29 +7,32 @@
     "markdox",
     "markdown",
     "documentation",
-    "gulp"
+    "gulp",
+    "ejs",
+    "dox"
   ],
   "author": {
```
Comment (gberger) [inline package.json]: you can add yourself as author/contributor @muroc

**Comment by gberger** [comment]
Looks great, see comments. Feel free to marge, and in the future commit directly as well (you're collab)

**Comment by muroc** [comment]
@gberger Many thanks for the review. Do you think you could prepare a new release (I suggest ⬆️ 2.0)?

**Comment by mchalapuk** [comment]
gulp-markdox2 deprecated on npm

**Pull request merged**